### PR TITLE
Fixes copy-paste issue in visual script editor

### LINKED ISSUE #41156: Copy and paste shortcuts don't work in Visual Script editor
**Godot version:**

3.2.3 rc3

**Issue description:**

When in Visual Script editor, you can't use Copy and Paste shortcuts. You need to select them from the Edit menu. All other shortcuts seem to work (including Cut and the unlisted Duplicate shortcut...)

### PR DESCRIPTION
**Pull request by Kartik1397**
Fixes #41156
Peek 2020-09-18 01-08

### CODE REVIEW

**Comment by fire** [comment]
Can you describe the implementation details? I see that you've moved the entire block into a function. Will try compiling and seeing if it's fixed. However, a comment would be helpful.

**Comment by EricEzaM** [comment]
This amount of changes is completely unnecessary. As far as I can tell, all you need to do is:

```cpp
graph->connect("copy_nodes_request", callable_mp(this, &VisualScriptEditor::_menu_option), varray(EDIT_COPY_NODES));
graph->connect("paste_nodes_request", callable_mp(this, &VisualScriptEditor::_menu_option), varray(EDIT_PASE_NODES));
```

The logic already exists in the menu options, why duplicate it all in different methods?

**Comment by EricEzaM** [comment]
He just moved logic from _menu_option() for EDIT_COPY_NODES and EDIT_PASTE_NODES into their own methods. Seems fine to me. I solved this issue just by doing.

```cpp
graph->connect("copy_nodes_request", callable_mp(this, &VisualScriptEditor::_menu_option), varray(EDIT_COPY_NODES));
graph->connect("paste_nodes_request", callable_mp(this, &VisualScriptEditor::_menu_option), varray(EDIT_PASE_NODES));
```

But moving it away from menu option and into their own method is probably cleaner. It makes _menu_option easier to read and separates the cut/copy logic into their own methods which I think is a good change.

File: modules/visual_script/visual_script_editor.cpp [hunk 1]
```diff
@@ -1694,9 +1694,144 @@ String VisualScriptEditor::_validate_name(const String &p_name) const {
...
+	}
+}
+
 void VisualScriptEditor::_on_nodes_delete() {
 	// delete all the selected nodes
 
+	if (!script->has_function(default_func)) {
+		return;
+	}
+
```
Comment (EricEzaM) [inline modules/visual_script/visual_script_editor.cpp:1834]: Why was this added? Since you just moved the logic from _menu_option to their own methods, I don't see why this is needed? Unless it solves some other bug that you haven't mentioned?

Comment (Kartik1397) [reply modules/visual_script/visual_script_editor.cpp:1834]: I haven't seen any bug without this.
I added this because there was break statement under this condition in EDIT_CUT_NODE which means it also prevent it from executing _on_nodes_delete() thats why I added this.

**Comment by akien-mga** [comment]
CC @swarnimarun

**Comment by DavidCambre** [comment]
Hi, @Kartik1397 I Would love to see this PR merged.
So have been trying to rebase this PR whiteout success.
Are you able to rebase? Then I'l try to test it.

**Comment by akien-mga** [comment]
Superseded by #54564. Thanks for the contribution nevertheless!
